lib/fileutils.rb

```diff
@@ -18,6 +18,15 @@
 #
 # Namespace for several file utility methods for copying, moving, removing, etc.
 #
+# == Path Arguments
+#
+# Some methods in \FileUtils accept _path_ arguments,
+# which are interpreted as paths to filesystem entries:
+#
+# - If the argument is a string, that value is the path.
+# - If the argument has method +:to_path+, it is converted via that method.
+# - If the argument has method +:to_str+, it is converted via that method.
+#
```

Comment:
From what I can tell, this is true because this is implemented in the underlying implementation of `FileUtil` methods (e.g. `FileUtils.cd` by `Dir.chdir`, `FileUtils.mkdir` by `Dir.mkdir`, `FileUtils.link` by `File.link`). Maybe this should be mentioned in the `Dir` and `File` class docs and this links to that section instead.

Reply:
I think it's sometimes b/c of File or Dir, but I think not always.  Either way, I'd still like to characterize the args fully here.  Also agree that File and Dir should mention; I'll get to them soon.

Thoughts on examples using Pathname for the methods here?

Reply:
I think it's fairly self-explanatory how to use Pathname instead of strings. I'm not if it is necessary to have examples using Pathname.